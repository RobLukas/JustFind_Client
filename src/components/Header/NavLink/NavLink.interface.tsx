import { SvgIconTypeMap } from '@material-ui/core';
import { OverridableComponent } from '@material-ui/core/OverridableComponent';

interface NavLinkProps {
  icon: OverridableComponent<SvgIconTypeMap>;
  link: string;
  desc: string;
}

export default NavLinkProps;
