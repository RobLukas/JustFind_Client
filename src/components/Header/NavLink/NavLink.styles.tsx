import styled from 'styled-components';
import { SvgIcon } from '@material-ui/core';

const NavLink = styled.a`
  @media (max-width: 1020px) {
    display: none;
  }
  margin: 0 20px 0 20px;
  display: flex;
  align-items: center;
  text-decoration: none;
  color: ${({ theme }) => theme.styles.secondary.color};
  &:hover {
    .MuiSvgIcon-root {
      fill: #ff4081;
      transition: unset;
    }
    color: #ff4081;
  }
`;

const MuiIcon = styled(SvgIcon)`
  margin-right: 12px;
` as typeof SvgIcon;

export default { NavLink, MuiIcon };
