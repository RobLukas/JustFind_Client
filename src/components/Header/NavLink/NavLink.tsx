import React, { FC } from 'react';
import S from './NavLink.styles';
import NavLinkProps from './NavLink.interface';

const NavLink: FC<NavLinkProps> = ({ icon, link, desc }: NavLinkProps) => {
  return (
    <S.NavLink href={link}>
      <S.MuiIcon component={icon} />
      <span>{desc}</span>
    </S.NavLink>
  );
};

export default NavLink;
