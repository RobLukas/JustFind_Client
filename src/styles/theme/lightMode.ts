export default {
  name: 'Light Theme',
  styles: {
    primary: {
      background: '#f3f6f8',
      color: '#37474f',
    },
    secondary: {
      background: '#fff',
      color: '#99a1ab',
    },
    dropdown: {
      button: {
        hover: '#f3f6f8',
      },
    },
    scrollbar: {
      thumbBackground: '#c1c1c1',
      background: 'rgba(0,0,0,0.07)',
      hover: 'rgba(0,0,0,0.09)',
    },
    filter: {
      button: {
        color: '#777',
      },
      selected: {
        background: 'linear-gradient(to right, #ba68c8, #9c27b0)',
        color: '#fff',
      },
    },
    offer: {
      header: {
        title: '#fff',
        filter: {
          brightness: '1',
        },
        icons: {
          companyName: '#ff5252',
          companySize: '#fb8c00',
          companyEMPType: '#ab47bc',
          companySkill: '#66bb6a',
          new: '#448aff',
        },
      },
      desc: {
        seperateLine: {
          color: '#f3f6f8',
        },
        skillPoint: {
          selected: '#ff4081',
          unselected: '#393939',
        },
      },
    },
    offers: {
      newTitle: {
        background: '#d9ddfc',
        color: '#4256ef',
      },
      techTitle: {
        background: '#fff',
        border: '#e0e0e0',
      },
    },
  },
};
