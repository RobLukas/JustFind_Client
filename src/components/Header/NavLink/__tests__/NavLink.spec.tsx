import React from 'react';
import { ThemeProvider } from 'styled-components';
import { cleanup, render } from '@testing-library/react';
import 'jest-styled-components';
import { SvgIcon } from '@material-ui/core';

import lightMode from '@styles/theme/lightMode';
import NavLink from '../NavLink';
import NavLinkProps from '../NavLink.interface';

describe('<NavLink />', () => {
  afterEach(cleanup);
  it('render correctly', () => {
    const props: NavLinkProps = { icon: SvgIcon, desc: 'link desc', link: '/' };
    const { container } = render(
      <ThemeProvider theme={lightMode}>
        <NavLink {...props} />
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
