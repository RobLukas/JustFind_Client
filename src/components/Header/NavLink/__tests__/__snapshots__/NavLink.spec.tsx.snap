// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLink /> render correctly 1`] = `
.c0 {
  margin: 0 20px 0 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #99a1ab;
}

.c0:hover {
  color: #ff4081;
}

.c0:hover .MuiSvgIcon-root {
  fill: #ff4081;
  -webkit-transition: unset;
  transition: unset;
}

.c1 {
  margin-right: 12px;
}

@media (max-width:1020px) {
  .c0 {
    display: none;
  }
}

<a
  class="c0"
  href="/"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-root c1"
    focusable="false"
    viewBox="0 0 24 24"
  />
  <span>
    link desc
  </span>
</a>
`;
